### PR TITLE
feat: add initial collection resource list page

### PR DESCRIPTION
**Pull request by seaerchin**
### TL;DR

This pull request introduces new functionality for creating and managing collection pages within the Studio application.

### What changed?
- Created `CreateCollectionPageModal` and associated components (`DetailsScreen`, `TypeScreen`, etc.) for adding new collection items.
- Enhanced server-side functionality by adding new API endpoints to handle collection pages (`collection.router.ts`, `collection.service.ts`).
- Adjusted various other components and utilities to accommodate these changes.

### How to test?
1. Start the Studio application.
2. Navigate to the collections section to see the new banner and table layouts.
3. Use the

### CODE REVIEW

**Comment by seaerchin** [comment]
* **#470** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/470?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#469** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/469?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#468** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @seaerchin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by karrui — APPROVED** [review]

**Comment by seaerchin** [comment]
### Merge activity

* **Aug 19, 5:49 AM EDT**: @seaerchin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/468).
